android/libraries/rib-base/src/main/java/com/badoo/ribs/android/RequestCodeBasedEventStreamImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.badoo.ribs.android
+
+import com.badoo.ribs.android.RequestCodeBasedEventStream.RequestCodeBasedEvent
+import com.badoo.ribs.core.Identifiable
+import com.badoo.ribs.core.requestcode.RequestCodeRegistry
+import com.jakewharton.rxrelay2.PublishRelay
+import com.jakewharton.rxrelay2.Relay
+import io.reactivex.Observable
+
+abstract class RequestCodeBasedEventStreamImpl<T : RequestCodeBasedEvent>(
```

Comment:
Not sure it makes sense to keep abstract `Impl` and the interface.
Right now you have bunch of other `Impl`s extending this one and interfaces extending base one.
Can we make it less scattered somehow?

Reply:
I don't see a way right now to do that (interfaces extend each other, and so do implementations). If you have any concrete suggestions, please tell it.

Reply:
No suggestions - I'm leaving as is for now